### PR TITLE
EOS-16642 Add option '-d' for 'hctl status' to add devices info in ou…

### PR DESCRIPTION
**Pull request by SwapnilGaonkar7**
Added new options '-d' and '--devices' to include devices info in 'hctl status' output
By default devices info will not be included in 'hctl status' output

Output:
-bash-4.2$ hctl status -d
Data pool:
    # fid name
    0x6f00000000000001:0x2f 'the pool'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x4f 'default': 'the pool' None None
Services:
    localhost  (RC)
    [started]  hax        0x7200000000000001:0x6   192.168.9.198@tcp:12345:1:1
    [started]  confd      0x7200000000000001:0x9   192.168.9.198@tcp:12345:2:1
    [started]  ioservice  0x7200000000000001:0xc   192.168.9.198@tcp:12345:2:2
    [unknown]  m0_client  0x7200000000000001:0x29  192.168.9.198@tcp:12345:4:1
    [unknown]  m0_client  0x7200000000000001:0x2c  192.168.9.198@tcp:12345:4:2
Devices:
    localhost
    [M0_NC_UNKNOWN]  /dev/null
    [M0_NC_UNKNOWN]  /dev/loop1
    [M0_NC_UNKNOWN]  /dev/loop2
    [M0_NC_UNKNOWN]  /dev/loop3
    [M0_NC_UNKNOWN]  /dev/loop4
    [M0_NC_UNKNOWN]  /dev/loop5
    [M0_NC_UNKNOWN]  /dev/loop6
    [M0_NC_UNKNOWN]  /dev/loop7
    [M0_NC_UNKNOWN]  /dev/loop8
    [M0_NC_UNKNOWN]  /dev/loop9
    [M0_NC_UNKNOWN]  /dev/loop0
-bash-4.2$ hctl status
Data pool:
    # fid name
    0x6f00000000000001:0x2f 'the pool'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x4f 'default': 'the pool' None None
Services:
    localhost  (RC)
    [started]  hax        0x7200000000000001:0x6   192.168.9.198@tcp:12345:1:1
    [started]  confd      0x7200000000000001:0x9   192.168.9.198@tcp:12345:2:1
    [started]  ioservice  0x7200000000000001:0xc   192.168.9.198@tcp:12345:2:2
    [unknown]  m0_client  0x7200000000000001:0x29  192.168.9.198@tcp:12345:4:1
    [unknown]  m0_client  0x7200000000000001:0x2c  192.168.9.198@tcp:12345:4:2
-bash-4.2$ hctl status -d --json
{
  "pools": [
    {
      "fid": "0x6f00000000000001:0x2f",
      "name": "the pool"
    }
  ],
  "profiles": [
    {
      "fid": "0x7000000000000001:0x4f",
      "name": "default",
      "pools": [
        "the pool",
        null,
        null
      ]
    }
  ],
  "filesystem": {
    "stats": {
      "fs_free_seg": 8590389096,
      "fs_total_seg": 8590951472,
      "fs_free_disk": 104689827840,
      "fs_avail_disk": 104689827840,
      "fs_total_disk": 104689827840,
      "fs_svc_total": 2,
      "fs_svc_replied": 2
    },
    "timestamp": 1611841778.130103,
    "date": "2021-01-28T06:49:38.130103"
  },
  "nodes": [
    {
      "name": "localhost",
      "svcs": [
        {
          "name": "hax",
          "fid": "0x7200000000000001:0x6",
          "ep": "192.168.9.198@tcp:12345:1:1",
          "status": "started"
        },
        {
          "name": "confd",
          "fid": "0x7200000000000001:0x9",
          "ep": "192.168.9.198@tcp:12345:2:1",
          "status": "started"
        },
        {
          "name": "ioservice",
          "fid": "0x7200000000000001:0xc",
          "ep": "192.168.9.198@tcp:12345:2:2",
          "status": "started"
        },
        {
          "name": "m0_client",
          "fid": "0x7200000000000001:0x29",
          "ep": "192.168.9.198@tcp:12345:4:1",
          "status": "unknown"
        },
        {
          "name": "m0_client",
          "fid": "0x7200000000000001:0x2c",
          "ep": "192.168.9.198@tcp:12345:4:2",
          "status": "unknown"
        }
      ]
    }
  ],
  "devices": [
    {
      "node_name": "localhost",
      "devs": [
        {
          "name": "/dev/null",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop1",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop2",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop3",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop4",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop5",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop6",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop7",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop8",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop9",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop0",
          "status": "M0_NC_UNKNOWN"
        }
      ]
    }
  ]
}
-bash-4.2$ hctl status --json
{
  "pools": [
    {
      "fid": "0x6f00000000000001:0x2f",
      "name": "the pool"
    }
  ],
  "profiles": [
    {
      "fid": "0x7000000000000001:0x4f",
      "name": "default",
      "pools": [
        "the pool",
        null,
        null
      ]
    }
  ],
  "filesystem": {
    "stats": {
      "fs_free_seg": 8590389096,
      "fs_total_seg": 8590951472,
      "fs_free_disk": 104689827840,
      "fs_avail_disk": 104689827840,
      "fs_total_disk": 104689827840,
      "fs_svc_total": 2,
      "fs_svc_replied": 2
    },
    "timestamp": 1611841778.130103,
    "date": "2021-01-28T06:49:38.130103"
  },
  "nodes": [
    {
      "name": "localhost",
      "svcs": [
        {
          "name": "hax",
          "fid": "0x7200000000000001:0x6",
          "ep": "192.168.9.198@tcp:12345:1:1",
          "status": "started"
        },
        {
          "name": "confd",
          "fid": "0x7200000000000001:0x9",
          "ep": "192.168.9.198@tcp:12345:2:1",
          "status": "started"
        },
        {
          "name": "ioservice",
          "fid": "0x7200000000000001:0xc",
          "ep": "192.168.9.198@tcp:12345:2:2",
          "status": "started"
        },
        {
          "name": "m0_client",
          "fid": "0x7200000000000001:0x29",
          "ep": "192.168.9.198@tcp:12345:4:1",
          "status": "unknown"
        },
        {
          "name": "m0_client",
          "fid": "0x7200000000000001:0x2c",
          "ep": "192.168.9.198@tcp:12345:4:2",
          "status": "unknown"
        }
      ]
    }
  ]
}
-bash-4.2$ hctl status --devices --json
{
  "pools": [
    {
      "fid": "0x6f00000000000001:0x2f",
      "name": "the pool"
    }
  ],
  "profiles": [
    {
      "fid": "0x7000000000000001:0x4f",
      "name": "default",
      "pools": [
        "the pool",
        null,
        null
      ]
    }
  ],
  "filesystem": {
    "stats": {
      "fs_free_seg": 8590389096,
      "fs_total_seg": 8590951472,
      "fs_free_disk": 104689827840,
      "fs_avail_disk": 104689827840,
      "fs_total_disk": 104689827840,
      "fs_svc_total": 2,
      "fs_svc_replied": 2
    },
    "timestamp": 1611841778.130103,
    "date": "2021-01-28T06:49:38.130103"
  },
  "nodes": [
    {
      "name": "localhost",
      "svcs": [
        {
          "name": "hax",
          "fid": "0x7200000000000001:0x6",
          "ep": "192.168.9.198@tcp:12345:1:1",
          "status": "started"
        },
        {
          "name": "confd",
          "fid": "0x7200000000000001:0x9",
          "ep": "192.168.9.198@tcp:12345:2:1",
          "status": "started"
        },
        {
          "name": "ioservice",
          "fid": "0x7200000000000001:0xc",
          "ep": "192.168.9.198@tcp:12345:2:2",
          "status": "started"
        },
        {
          "name": "m0_client",
          "fid": "0x7200000000000001:0x29",
          "ep": "192.168.9.198@tcp:12345:4:1",
          "status": "unknown"
        },
        {
          "name": "m0_client",
          "fid": "0x7200000000000001:0x2c",
          "ep": "192.168.9.198@tcp:12345:4:2",
          "status": "unknown"
        }
      ]
    }
  ],
  "devices": [
    {
      "node_name": "localhost",
      "devs": [
        {
          "name": "/dev/null",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop1",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop2",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop3",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop4",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop5",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop6",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop7",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop8",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop9",
          "status": "M0_NC_UNKNOWN"
        },
        {
          "name": "/dev/loop0",
          "status": "M0_NC_UNKNOWN"
        }
      ]
    }
  ]
}
-bash-4.2$ hctl status --devices
Data pool:
    # fid name
    0x6f00000000000001:0x2f 'the pool'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x4f 'default': 'the pool' None None
Services:
    localhost  (RC)
    [started]  hax        0x7200000000000001:0x6   192.168.9.198@tcp:12345:1:1
    [started]  confd      0x7200000000000001:0x9   192.168.9.198@tcp:12345:2:1
    [started]  ioservice  0x7200000000000001:0xc   192.168.9.198@tcp:12345:2:2
    [unknown]  m0_client  0x7200000000000001:0x29  192.168.9.198@tcp:12345:4:1
    [unknown]  m0_client  0x7200000000000001:0x2c  192.168.9.198@tcp:12345:4:2
Devices:
    localhost
    [M0_NC_UNKNOWN]  /dev/null
    [M0_NC_UNKNOWN]  /dev/loop1
    [M0_NC_UNKNOWN]  /dev/loop2
    [M0_NC_UNKNOWN]  /dev/loop3
    [M0_NC_UNKNOWN]  /dev/loop4
    [M0_NC_UNKNOWN]  /dev/loop5
    [M0_NC_UNKNOWN]  /dev/loop6
    [M0_NC_UNKNOWN]  /dev/loop7
    [M0_NC_UNKNOWN]  /dev/loop8
    [M0_NC_UNKNOWN]  /dev/loop9
    [M0_NC_UNKNOWN]  /dev/loop0

### CODE REVIEW

**Comment by SwapnilGaonkar7** [comment]
@mssawant @knekrasov Please review.

File: utils/hare-status [hunk 1]
```diff
@@ -304,6 +320,10 @@ def main(argv=None):
         print('Cluster is not running', file=sys.stderr)
         return 1
 
+    if opts.devices:
+        global devicesRequested
```
Comment (knekrasov) [inline utils/hare-status:324]: global flag is not the best solution

File: utils/hare-status [hunk 2]
```diff
@@ -229,16 +231,26 @@ def devices(cns: Consul, node_name: str) -> List[Device]:
...
-        'pools': [x for x in sns_pools(cns)],
-        'profiles': [{'fid': x.fid, 'name': x.name, 'pools': x.pools}
-                     for x in profiles(cns)],
-        'filesystem': get_fs_stats(cns),
-        'nodes': [Node(name=h, svcs=processes(cns, h))
-                  for h in node_names(cns)],
-        'devices': [Devices(node_name=h, devs=devices(cns, h))
-                    for h in node_names(cns)]
-    }
+    if devicesRequested:
```
Comment (knekrasov) [inline utils/hare-status:234]: The dicts in `if` and `else` clauses differ with one key only. Please think of the ways how to make the code more readable (e.g. why do we need `else`?).

File: utils/hare-status (hunk 1 above)
Comment (knekrasov) [reply utils/hare-status:324]: Note: camelCase is not very common for Python. 
Suggestion: `s/devicesRequested/devices_requested/g`

**Review by knekrasov — CHANGES_REQUESTED** [review]

File: utils/hare-status (hunk 2 above)
Comment (SwapnilGaonkar7) [reply utils/hare-status]: Done

File: utils/hare-status (hunk 1 above)
Comment (SwapnilGaonkar7) [reply utils/hare-status]: Done

**Comment by SwapnilGaonkar7** [comment]
@knekrasov @mssawant Please have a look

**Review by mssawant — APPROVED** [review]
Thank you.

**Review by knekrasov — APPROVED** [review]

**Pull request merged**